src/modules/emote_menu/components/Menu.jsx

```diff
@@ -0,0 +1,146 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import Divider from 'rsuite/lib/Divider/index.js';
+import keycodes from '../../../utils/keycodes.js';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/menu.module.css';
+import Emotes from './Emotes.jsx';
+import Header from './Header.jsx';
+import Preview from './Preview.jsx';
+import Sidebar from './Sidebar.jsx';
+
+export default function EmoteMenu({triggerRef, appendToChat}) {
+  const [search, setSearch] = useState('');
+  const [selected, setSelected] = useState(null);
+  const [, setUpdated] = useState(false);
+
+  const [altPressed, setAltPressed] = useState(false);
+  const [shiftPressed, setShiftPressed] = useState(false);
+
+  const [section, setSection] = useState({
+    eventKey: null,
+    scrollTo: false,
+  });
+
+  const [arrowKeys, setArrowKeys] = useState({
+    top: false,
+    down: false,
+    right: false,
+    left: false,
+  });
+
+  const onHide = useCallback(() => triggerRef.current.close(), [triggerRef]);
+
+  const handleClick = useCallback(
+    (emote) => {
+      if (altPressed) {
+        emoteStore.toggleFavorite(emote);
+        return;
+      }
+
+      appendToChat(emote.code);
+      emoteStore.trackHistory(emote);
+
+      if (shiftPressed) {
+        return;
+      }
+
+      onHide();
+    },
+    [altPressed, shiftPressed]
+  );
+
+  useEffect(() => setSearch(''), [section]);
+
+  const handleKeyDownCallback = useCallback(
+    (event) => {
+      if (event.keyCode === keycodes.Enter) {
+        event.preventDefault();
+        handleClick(selected);
+      }
+    },
+    [selected]
+  );
+
+  useEffect(() => {
+    function buttonPressCallback(event) {
+      setAltPressed(event.altKey);
+      setShiftPressed(event.shiftKey);
+    }
+
+    function handleArrowKeys(event) {
+      if (
+        event.keyCode === keycodes.UpArrow ||
+        event.keyCode === keycodes.DownArrow ||
+        event.keyCode === keycodes.RightArrow ||
+        event.keyCode === keycodes.LeftArrow
+      ) {
+        setArrowKeys({
```

Comment:
I think I commented previously we should stop storing state for these. instead, either add a listener where we need them or pass a callback function in that allows the nested component to hook into these events on mount